src/phone_number/landline.rs

```diff
@@ -1,17 +1,42 @@
 use lazy_static::lazy_static;
 use regex::Regex;
 
+use crate::province::{IranProvince, PROVINCES};
+
 lazy_static! {
     static ref LANDLINE_NUMBER_REGEX: Regex =
         Regex::new(r"(\+98|0|98|0098)?([1-9]{2})+(\d{8})$").unwrap();
```

Comment:
It will match all the mobile phone numbers. I think the limitation is the 9 as the very first digit!
So this regex is also wrong!
I can see the same `$` and `^` problem here too.

Reply:
So I think it makes more sense: `^(\+98|0|98|0098)([1-8]{2})(\d{8})$`
At least it passes the current tests.